src/main.cpp

```diff
@@ -0,0 +1,206 @@
+#include <vector>
+#include <iostream>
+#include <string>
+#include <windows.h>
+
+#include "pugixml.hpp"
+
+#define SDL_MAIN_HANDLED
+#include <SDL.h>
+
+const int screen_width = GetSystemMetrics(SM_CXSCREEN);
+const int screen_height = GetSystemMetrics(SM_CYSCREEN);
+
+enum StatusCode {
+    SUCCESS = 0,
+    FILENAME_NOT_PROVIDED,
+    PARSER_FAILURE,
+    SDL_INIT_FAILURE
+};
+
+const char *status_messages[] = {
+        "Success\n",
+        "Filename was not provided\n",
+        "Parser failure\n",
+        "SDL could not be initialized\n"
+};
+
+const char *parser_e_id = "e_id";
+const char *parser_c_id = "c_id";
+const char *parser_x = "x";
+const char *parser_y = "y";
+const char *parser_height = "height";
+const char *parser_width = "width";
+const char *parser_end_elem = "end_element";
+
+struct NormalizedPoint {
+    float n_x;
+    float n_y;
+
+    NormalizedPoint() : n_x(0), n_y(0) {}
+};
+
+struct NormalizedConnection {
+    unsigned int id;
+    unsigned int start_element_id;
+    unsigned int end_element_id;
+    std::vector<NormalizedPoint> vertices;
+
+    NormalizedConnection() : id(-1), start_element_id(-1), end_element_id(-1) {}
+};
+
+struct NormalizedElement {
+    unsigned int id;
+    NormalizedPoint point;
+    float n_w;
+    float n_h;
+    std::vector<NormalizedConnection> connections;
+
+    NormalizedElement() : id(-1), n_w(0), n_h(0) {}
+
+    friend std::ostream &operator<<(std::ostream &out, const NormalizedElement &element_to_print);
```

Comment:
`operator<<` overload does not have to be a `friend` since all `NormalizedElement` fields are public.